src/routers/institutions.py

```diff
@@ -105,6 +108,30 @@ async def get_institution(
     return res
 
 
+@router.get("/{lei}/types/{type}", response_model=VersionedData[List[SblTypeAssociationDetailsDto]] | None)
+@requires("authenticated")
+async def get_types(request: Request, lei: str, type: InstitutionType):
+    match type:
+        case "sbl":
+            fi = await repo.get_institution(request.state.db_session, lei)
+            return VersionedData(version=fi.version, data=fi.sbl_institution_types) if fi else None
+        case "hmda":
+            raise HTTPException(status_code=HTTPStatus.NOT_IMPLEMENTED, detail="HMDA type not yet supported")
+
+
+@router.put("/{lei}/types/{type}", response_model=VersionedData[List[SblTypeAssociationDetailsDto]] | None)
+@requires("authenticated")
+async def update_types(request: Request, lei: str, type: InstitutionType, types_patch: SblTypeAssociationPatchDto):
+    match type:
+        case "sbl":
+            fi = await repo.update_sbl_types(
+                request.state.db_session, request.user, lei, types_patch.sbl_institution_types
+            )
+            return VersionedData(version=fi.version, data=fi.sbl_institution_types) if fi else None
+        case "hmda":
+            raise HTTPException(status_code=HTTPStatus.NOT_IMPLEMENTED, detail="HMDA type not yet supported")
+
+
```

Comment:
Just the question I put in Mattermost when you lost power:

Wondering if we should be returning 204 NO_CONTENT instead of None?  I've done that with a few of the endpoints in filing, and added it to the wiki for others to bring up if we want to decide on a consistent approach

Thoughts?

Reply:
yeap, the sbl case I've updated to use no_content instead

Reply:
hmda one is a question mark at the moment; if we do allow more open modifications later down the line, I'm guessing the hmda one will be more like the other normal fields, rather than the special case types dealing